### PR TITLE
[Added][BEL-1544] Limit param was removed in favor to pass object param

### PR DESCRIPTION
**Pull request by benitocas88**
# Why

- Currently is not possible get data using filter options in node-sdk using `list` function.
- Receiving an Object is more flexible and scalable

# What?
- Allow customers get only data he wants sending parameter options to filter

### CODE REVIEW

File: src/resources.js [hunk 1]
```diff
@@ -14,11 +16,22 @@ class Resource {
...
+    let stop = limit;
+    let params = filters;
+
+    if (typeof stop === 'object') {
+      params = stop;
+      stop = LIST_LIMIT_DEFAULT;
+    }
+
+    const qs = fromObjectQueryParams(params);
+    const ep = `${this.#endpoint}${qs}`;
```
Comment (shekodn) [inline src/resources.js:33]:
```suggestion
    const endpoint = `${this.#endpoint}${qs}`;
```

File: src/resources.js [hunk 2]
```diff
@@ -14,11 +16,22 @@ class Resource {
...
+    const result = await this.session.list(ep, stop);
```
Comment (shekodn) [inline src/resources.js:34]:
```suggestion
    const result = await this.session.list(endpoint, stop);
```

File: src/resources.js [hunk 3]
```diff
@@ -14,11 +16,22 @@ class Resource {
...
    * @throws {RequestError}
    */
-  async list(limit = 100) {
-    const result = await this.session.list(this.#endpoint, limit);
+  async list(limit = LIST_LIMIT_DEFAULT, filters) {
+    let stop = limit;
+    let params = filters;
+
+    if (typeof stop === 'object') {
+      params = stop;
```
Comment (shekodn) [inline src/resources.js:28]: why do we use the variable `stop`? could we just use `limit`?

**Review by shekodn — COMMENTED** [review]
I left a couple of changes/suggestions. Let me know what you think :)

File: src/utils.js [hunk 4]
```diff
@@ -8,5 +8,13 @@ const encodeFileB64 = (path) => {
   return null;
 };
 
+const fromObjectQueryParams = (obj) => {
+  if (obj && obj.constructor === Object) {
+    return `?${stringify(obj)}`;
+  }
+  return '';
+};
+
```
Comment (menecio) [inline src/utils.js:17]: There is no need to build the query parameters by yourself, axios already provides `params` for that, basically what you need to make this https://github.com/belvo-finance/belvo-js/blob/master/src/http.js#L77 able to receive the filters and send them using params.

**Review by menecio — CHANGES_REQUESTED** [review]
Welcome Beni! I'm missing a test case for the filtering here and I left a comment

File: src/resources.js [hunk 5]
```diff
@@ -14,11 +16,22 @@ class Resource {
...
    * @returns {array} List of results.
    * @throws {RequestError}
    */
-  async list(limit = 100) {
-    const result = await this.session.list(this.#endpoint, limit);
+  async list(limit = LIST_LIMIT_DEFAULT, filters) {
+    let until = limit;
+    let params = filters;
+
+    if (typeof limit === 'object') {
```
Comment (benitocas88) [inline src/resources.js:27]: @menecio what do you think about this case, preventing `limit` override if users send object insted of `limit`:
`client.[resource].list({ amount__gte: 5 })`
If they want use filter the `limit` param would be mandatory?

**Comment by benitocas88** [comment]
This is a doc example with the change

image

File: tests/transactions.test.js [hunk 6]
```diff
@@ -196,3 +209,17 @@ test('can resume transaction session', async () => {
   expect(result).toEqual(transaction);
   expect(mocker.scope.isDone()).toBeTruthy();
 });
+
+test('can list transactions given filters', async () => {
```
Comment (benitocas88) [inline tests/transactions.test.js:213]: @menecio I wanted to do this test at least for some resource to ensure correct functionality

**Review by gingrassia — APPROVED** [review]

**Review by menecio — APPROVED** [review]

**Pull request merged**